tests/orquestra/quantum/circuits/_itertools_test.py

```diff
@@ -139,3 +140,50 @@ def test_counts_of_combined_measurements_are_correct(
                 combined_counts,
             )
         )
+
+
+class TestCombiningBitstrings:
+    def test_raises_error_when_multiplicities_dont_match_measurements(self):
+        multiplicities = [1, 2, 3, 2, 2]
+        # Clearly a mismatch, we should have 10 bitstring sequences
+        measurements = [["00", "11"] for _ in range(5)]
+
+        with pytest.raises(ValueError):
+            combine_measurement_counts(measurements, multiplicities)
+
+    @pytest.mark.parametrize(
+        "all_bitstrings, multiplicities, combined_bitstrings",
+        [
+            ([["00", "11"]], [1], [["00", "11"]]),
+            (
+                [
+                    ["00", "11"],
+                    ["01", "00"],
+                    ["00", "00"],
+                    ["000"],
+                    ["111", "001"],
+                    ["00000"],
+                    ["0", "0", "0"],
+                    ["1", "0"],
+                    ["0", "0"],
+                ],
+                [3, 2, 1, 3],
+                [
+                    ["00", "11", "01", "00", "00", "00"],
+                    ["000", "111", "001"],
+                    ["00000"],
+                    ["0", "0", "0", "1", "0", "0", "0"],
+                ],
+            ),
+        ],
+    )
+    def test_counts_of_combined_bitstrings_are_correct(
+        self, all_bitstrings, multiplicities, combined_bitstrings
+    ):
+        assert all(
+            actual == expected
+            for actual, expected in zip(
+                combine_bitstrings(all_bitstrings, multiplicities),
+                combined_bitstrings,
+            )
+        )
```

Comment:
[suggestion]: Do you think it would be worth it to make more, simpler tests here? Simpler tests are easier to write and might make bugs more easy to decipher.

Reply:
Do you have anything simpler in mind? It seems to me that this is the simplest possible test, where we check the return value of the function under test (`combine_bitstrings`)  against the expected, correct return value. I don't know anything simpler, but I'll welcome any suggestion :slightly_smiling_face: